specs/error_state/ErrorOutOfGasSLOAD.md

```diff
@@ -0,0 +1,41 @@
+# ErrorOutOfGasSLOAD state
+
+## Procedure
+
+Handle the corresponding out of gas error for opcode `SLOAD`.
+
+### EVM behavior
+
+For the current [go-ethereum] code, the out of gas error may occur for [constant gas] or [dynamic gas].
+
+The [default SLOAD constant gas] is set to `50` and no dynamic gas.
+When [EIP-150] is enabled, [EIP-150 SLOAD constant gas] is set as `200`.
+When [EIP-2929] is enabled, [EIP-2929 SLOAD dynamic gas] function is set.
+
+For this gadget, the gas cost is calculated with [EIP-2929] as:
+```
+gas_cost = GAS_COST_WARM_ACCESS if is_warm_access else COLD_SLOAD_COST
+```
+
+### Constraints
+
+1. Constrain `gas_left < gas_cost`.
+2. Current call must fail.
+3. If it's a root call, it transits to `EndTx`.
+4. if it isn't a root call, it restores caller's context by reading to `rw_table`, then does step state transition to it.
+
+### Lookups
+
+1. 4 call context lookups for `tx_id`, `callee_address`, `is_success` and `rw_counter_end_of_reversion`.
+2. 1 stack read for `storage_key` (no stack write for `storage_value` when out of gas).
+3. 1 account storage access list read for `storage_key`.
+
+[//]: # "badges"
+[go-ethereum]: https://github.com/ethereum/go-ethereum
```

Comment:
remove these links ?

Reply:
Deleted links in [aa8d6f1](https://github.com/scroll-tech/zkevm-specs/pull/57/commits/aa8d6f1ec6490a6129603bbe7a06a8c9b7d5ace5).